odxtools/communicationparameter.py

```diff
@@ -1,10 +1,34 @@
 # SPDX-License-Identifier: MIT
 # Copyright (c) 2022 MBition GmbH
 
-from typing import Optional, List
+from typing import Optional, List, TYPE_CHECKING
+import warnings
 
-from .odxlink import OdxLinkRef, OdxDocFragment
+from odxtools.exceptions import OdxWarning
+
+from .odxlink import OdxLinkDatabase, OdxLinkRef, OdxDocFragment
 from .utils import read_description_from_odx
+from .comparam_subset import BaseComparam, Comparam, ComplexComparam, read_complex_value_from_odx
+
+
+def _get_comparam_value(comparam: BaseComparam, value):
+    if not comparam:
+        return value
+    if value is None:
+        value = comparam.physical_default_value
+
+    if isinstance(comparam, Comparam):
+        return comparam.dop.physical_type.base_data_type.from_string(value)
+
+    if isinstance(comparam, ComplexComparam) and isinstance(value, list):
+        value = value + [None] * (len(comparam.comparams) - len(value))
+        result = dict()
+        for i, cp in enumerate(comparam.comparams):
+            result[cp.short_name] = _get_comparam_value(cp, value[i])
+        return result
```

Comment:
@andlaus I am not 100% sure, since I don't have a PDX to check on, but this should remove the need for "candela workarounds"

Reply:
if possible, I'm all for it. When we implemented this, we noticed that the files written by candela use a different field for the can IDs than specified by the `.odx-cs` fragments shipped with the standard. (I think that I also looked up if the `pdx` files were internally and I thought it wasn't, but since I did this manually, I probably screwed up while doing this...)